nemo/collections/asr/models/asrconvctcmodel.py

```diff
@@ -0,0 +1,237 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Iterable, List, Optional
+
+import nemo
+from nemo import logging
+from nemo.core import NeMoModel, NeuralGraph, NeuralModule, NeuralType, OperationMode, PretrainedModelInfo
+from nemo.utils import maybe_download_from_cloud
+
+
+class ASRConvCTCModel(NeMoModel):
+    """
+    Generic convolutional CTC-based model with encoder and decoder. It also contains pre-processing module and
+    data augmentation model.
+
+    Example models of this type are: JasperNet and QuartzNet
+    """
+
+    def __init__(
+        self,
+        preprocessor_params: Dict,
+        encoder_params: Dict,
+        decoder_params: Dict,
+        spec_augment_params: Optional[Dict] = None,
+    ):
+        super().__init__()
+        # Instantiate necessary modules
+        self.__vocabulary = None
+        preprocessor, spec_augmentation, encoder, decoder = self.__instantiate_modules(
+            preprocessor_params, encoder_params, decoder_params, spec_augment_params
+        )
+        self._operation_mode = OperationMode.training
+
+        # self.__training_neural_graph = NeuralGraph(operation_mode=OperationMode.training)
+        self.__training_neural_graph = NeuralGraph(operation_mode=OperationMode.both)
+        with self.__training_neural_graph:
+            # Copy one input port definitions - using "user" port names.
+            self.__training_neural_graph.inputs["input_signal"] = preprocessor.input_ports["input_signal"]
+            self.__training_neural_graph.inputs["length"] = preprocessor.input_ports["length"]
+            # Bind the selected inputs. Connect the modules
+            i_processed_signal, i_processed_signal_len = preprocessor(
+                input_signal=self.__training_neural_graph.inputs["input_signal"],
+                length=self.__training_neural_graph.inputs["length"],
+            )
+            if spec_augmentation is not None:
+                i_processed_signal = spec_augmentation(input_spec=i_processed_signal)
+            i_encoded, i_encoded_len = encoder(audio_signal=i_processed_signal, length=i_processed_signal_len)
+            i_log_probs = decoder(encoder_output=i_encoded)
+            # Bind the selected outputs.
+            self.__training_neural_graph.outputs["log_probs"] = i_log_probs
+            self.__training_neural_graph.outputs["encoded_len"] = i_encoded_len
+
+        # self.__evaluation_neural_graph = NeuralGraph(operation_mode=OperationMode.evaluation)
+        self.__evaluation_neural_graph = NeuralGraph(operation_mode=OperationMode)
```

Comment:
Is there a reason why operation mode is `OperationMode`?

Reply:
bug. fixed